scripts/dts/extract_dts_includes.py

```diff
@@ -426,9 +426,31 @@ def yaml_inc_error(msg):
     raise yaml.constructor.ConstructorError(None, None, msg)
 
 
+def extract_cpu_clock_frequencies():
```

Comment:
A question: why don't we extract `clock-frequency` property via a .yaml binding file?

Reply:
The yaml file is parsed only if there is a "compatible" property, which is omitted in this node in most of the files. Also, such approach would require plenty of yaml files as we have plenty of different CPUs supported in Zephyr.

Reply:
We should add the YAML files for the CPUs.  There can't be that many.

Reply:
We have 8 architectures in Zephyr. Only on ARM we have Cortex-M0, M0+, M3, M4, M4F, M7 and M33F. If we assume that we have 6 different CPUs in given architecture in average, this will require 48 different YAML files...

Reply:
> We have 8 architectures in Zephyr. Only on ARM we have Cortex-M0, M0+, M3, M4, M4F, M7, M23 and M33F. If we assume that we have 6 different CPUs in given architecture in average, this will require 48 almost identical YAML files...

Then we have that many YAML files.  We can use the 'inherit' to reduce the duplication between them.

Reply:
Just for clarification. There will be dozens of files which will look like this:

```
---
title: ARM Cortex-M0 CPU
version: 0.1

description: ARM Cortex-M0 CPU binding

inherits:
    !include cpu-base.yaml

properties:
    compatible:
      constraint: "arm,cortex-m0"
...
```

```
---
title: ARM Cortex-M0+ CPU
version: 0.1

description: ARM Cortex-M0+ CPU binding

inherits:
    !include cpu-base.yaml

properties:
    compatible:
      constraint: "arm,cortex-m0+"
...
```

```
---
title: ARM Cortex-M3 CPU
version: 0.1

description: ARM Cortex-M3 CPU binding

inherits:
    !include cpu-base.yaml

properties:
    compatible:
      constraint: "arm,cortex-m3"
...
```

And this is what we want.
Am I correct?

Reply:
Yes

Reply:
Discussed on slack. This is the approach we want.

Reply:
@galak: I have tried to go in the direction you suggested, but this really does no have much sense for me.

1. The number of YAML files is considerable.
2. We also have to introduce plenty of dts_fixup.h files, as defines generated have part of compatible string in the name.
3. For Cortex-M0+ we are getting `DT_ARM_CORTEX_M0+_0_CLOCK_FREQUENCY` define. So scripts modification is also needed to avoid `+` character (I assume that we will not change current compatible strings using `+` character).

Summarising: The effort required to handle this in the way you proposed is really big and mostly consists of creating work-arounds in dts_fixup.h files...

Do you really would like to go in this direction?

Reply:
> @galak: I have tried to go in the direction you suggested, but this really does no have much sense for me.
> 
> 1. The number of YAML files is considerable.
> 2. We also have to introduce plenty of dts_fixup.h files, as defines generated have part of compatible string in the name.
> 3. For Cortex-M0+ we are getting `DT_ARM_CORTEX_M0+_0_CLOCK_FREQUENCY` define. So scripts modification is also needed to avoid `+` character (I assume that we will not change current compatible strings).
> 
> Summarising: The effort required to handle this in the way you proposed is really big and mostly consists of creating work-arounds in dts_fixup.h files...
> 
> Do you really would like to go in this direction?

Yes this is still the direction we should go.  I'd suggest we create a separate PR for the CPU node work.

Reply:
> @galak: I have tried to go in the direction you suggested, but this really does no have much sense for me.
> 
> 1. The number of YAML files is considerable.
> 2. We also have to introduce plenty of dts_fixup.h files, as defines generated have part of compatible string in the name.
> 3. For Cortex-M0+ we are getting `DT_ARM_CORTEX_M0+_0_CLOCK_FREQUENCY` define. So scripts modification is also needed to avoid `+` character (I assume that we will not change current compatible strings).
> 
> Summarising: The effort required to handle this in the way you proposed is really big and mostly consists of creating work-arounds in dts_fixup.h files...
> 
> Do you really would like to go in this direction?

Yes this is still the direction we should go.  I'd suggest we create a separate PR for the CPU node work.

Reply:
Ok. How to handle the `+` in `DT_ARM_CORTEX_M0+_0_CLOCK_FREQUENCY` issue? Should we replace `+` with a `PLUS` in define name or change compatible string to something else?

Reply:
@galak: The work you asked for is done. The CPU frequency is extracted through YAML files and dts_fixup.h is used to gather back the defines where such step is needed.

Please check if this PR is now OK for you.